### ISSUE TITLE
Macros with multi-line comments may lead to incorrect string parsing

### ISSUE THREAD

**Issue by Theldus** [ColinIanKing/kernelscan#1]
Hi,
While using kernelscan in the Linux src tree, I realized that there is a possibility of identifying wrong strings if there is a comment on the same line as a macro, which expands to several lines and has a quotation mark in the following lines, a minimal example could be:
```c
#define SOME_MACRO 0xbeef /* bad comment.
	" veri evil. */

const char *str = "wrongstring";
```
If I run kernelscan on this, I get the following output:
```text
$ ./kernelscan -l -c -d american-english ./test.c
const
str
veri

1 files scanned
1 lines scanned (0.000 Mbytes)
0 print statements found
102305 words and 168757 nodes in dictionary heap
869273 chars mapped to 18394513 bytes of heap, ratio=1:21.16
2163 printk style statements being searched
3 unique bad spellings found (3 non-unique)
scanned 5322.72 lines per second
(kernelscan 0.3.4)
```
When kernelscan runs `skip_macros`, it continues until it finds a line break, which leaves the rest of the comment on the next lines without being processed, so when it finds a ", it believes it is the beginning of a new string.

A real example could be find in the file [drivers/scsi/scsi_transport_spi.c:29](https://elixir.bootlin.com/linux/v5.11.1/source/drivers/scsi/scsi_transport_spi.c#L29).

(I don't know if this is already a known bug, or if there is a proper mailing list to post this on, I'm sorry if I'm doing it wrong)

**Comment by ColinIanKing** [comment]
Thanks for reporting this issue. This is indeed an new issue that I've not seen. I'll see if I can get a fix sorted out some time soon. Good catch by the way, and very helpful bug report.

**Comment by ColinIanKing** [comment]
I've pushed a fix to the repository, commit 1242dbd71

This should consume comments that are found inside macro #defines

**Comment by Theldus** [comment]
Thank you @ColinIanKing for addressing this so quickly, worked like a charm here =).
